### DISCUSSION TITLE
How to control the permissions

### DISCUSSION THREAD

**Discussion by duanyongyao** [milvus-io/milvus#6646] [Ideas & Feature requests]
How do I control the permissions, such as access via user name and password, and control read and write permissions

The python SDK documentation provides the following ways to connect to Milvus server,but I think this is very dangerous that anyone who knows the ip and port can have full permissions, Create,Read,Update,Delete.

```python
# Connect to Milvus server
milvus = Milvus(host='localhost', port='19530')
```

Is there any way I can control permissions and ensure data security?

**Comment by yhmo** [comment]
Currently milvus doesn't support user/account management, both for v1.x and v2.0. 
You need to build a middle ware to do it. The [Mishards](https://github.com/milvus-io/milvus/tree/1.1/shards) is a good example for such middle ware. 
The basic solution is:
- The middle ware implements same RPC interfaces with Milvus. The Mishards implements all Milvus interfaces in this class: https://github.com/milvus-io/milvus/blob/1.1/shards/mishards/service_handler.py, and inject the class into gRPC as a handler, the inject code is in this file(in the start() method): https://github.com/milvus-io/milvus/blob/1.1/shards/mishards/server.py
- The middle ware persist user/account information into MongoDB or SQLite, manage permissions.
- The middle ware manages a list of Milvus servers, distribute request from different user to different Milvus server.
- The middle ware doesn't take part in computing, it directly return results from Milvus server to user client.

**Reply by duanyongyao** [reply]
Thank you, I will try it and see if it solves the problem.
